packages/plugin-node/src/services/awsS3.ts

```diff
@@ -58,13 +60,19 @@ export class AwsS3Service extends Service implements IAwsS3Service {
             return false;
         }
 
-        this.s3Client = new S3Client({
+        const clientConfig: S3ClientConfig = {
             region: AWS_REGION,
             credentials: {
                 accessKeyId: AWS_ACCESS_KEY_ID,
                 secretAccessKey: AWS_SECRET_ACCESS_KEY,
             },
-        });
+        };
+        if (AWS_S3_ENDPOINT) {
```

Comment:
only add `endpoint` to the config if the env var is set, ensuring existing behavior with AWS-only endpoints occurs